wled00/FX_fcn.cpp

```diff
@@ -235,12 +233,12 @@ void WS2812FX::setPixelColor(uint16_t i, byte r, byte g, byte b, byte w)
     }
     uint32_t col = ((w << 24) | (r << 16) | (g << 8) | (b));
 
-    bool reversed = IS_REVERSE;
+    /* Set all the pixels in the group */
     uint16_t realIndex = realPixelIndex(i);
     uint16_t len = SEGMENT.length();
 
     for (uint16_t j = 0; j < SEGMENT.grouping; j++) {
-      int indexSet = realIndex + (reversed ? -j : j);
+      uint16_t indexSet = realIndex + (IS_REVERSE ? -j : j);
```

Comment:
Let's keep this `int`. Even though it makes no immediate logical sense, this safeguards against an effect function causing a crash on ESP32 by writing outside of segment bounds (https://github.com/Aircoookie/WLED/commit/789c00dde12634a9db415e012d06550f3d0370e5)

Reply:
Just realized this was `int16_t` before, so `uint16_t` might be fine.

Reply:
There should be other safeguards to prevent crossing segment bounds.